pallets/fiat_filter/src/lib.rs

```diff
@@ -0,0 +1,306 @@
+
+    /// The outer call (dispatchable) that this module is called with. It is possible to use another type here, but
+    /// it's expected that your runtime::Call will be used.
+    /// Since we need to match by origin crate, we implement `IsSubType` for each Call type we want to match for.
+    type Call: Parameter
+        + UnfilteredDispatchable<Origin = Self::Origin>
+        + GetDispatchInfo
+        + IsSubType<did::Call<Self>>
+        + IsSubType<anchor::Call<Self>>
+        + IsSubType<blob::Call<Self>>
+        + IsSubType<revoke::Call<Self>>
+        + IsSubType<attest::Call<Self>>;
+    /// The module's Currency type definition
+    type Currency: Currency<Self::AccountId>;
+}
+
+// The pallet's runtime storage items.
+decl_storage! {
+    // A unique name is used to ensure that the pallet's storage items are isolated.
+    // This name may be updated, but each pallet in the runtime must use a unique name.
+    // ----------------------------------vvvvvvvvvvvvvv
+    trait Store for Module<T: Config> as FiatFilterModule {
+        /// price of one DOCK in fiat (for now, only USD)
+        pub DockFiatRate get(fn dock_fiat_rate) config(): Permill = Permill::from_parts(22511);
+        /// price update frequency (in number of blocks)
+        pub UpdateFreq get(fn update_freq) config(): <T as system::Config>::BlockNumber = 10u32.into();
+        // /// block number of last DockUsdRate update
+        pub LastUpdatedAt get(fn last_updated_at): <T as system::Config>::BlockNumber;
+    }
+}
+
+// Pallets use events to inform users when important changes are made.
+decl_event! {
+    pub enum Event<T> where
+        <T as frame_system::Config>::BlockNumber,
+    {
+        /// on set_dock_usd_rate executed
+        /// event parameters: [new_dock_usd_rate]
+        DockFiatRateUpdated(Permill),
+        /// on root_set_update_freq executed
+        /// event parameters: [new_update_frequency_blocks]
+        UpdateFreqUpdated(BlockNumber),
+    }
+}
+
+// Errors inform users that something went wrong.
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// Failed calculation because of numeric type overflow
+        ArithmeticOverflow,
+        /// checked_div: tried to divide by zero
+        DivideByZero,
+        /// Call is not among the core_mods ones that should go through fiat_filter
+        UnexpectedCall,
+    }
+}
+
+// The module's callable functions (Dispatchable functions)
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: T::Origin {
+        // Errors must be initialized if they are used by the pallet.
+        type Error = Error<T>;
+        // Events must be initialized if they are used by the pallet.
+        fn deposit_event() = default;
+        /// `on_initialize` gets called on every new block
+        fn on_initialize(n: T::BlockNumber) -> Weight {
```

Comment:
Should I simply get the price pair with `T::PriceProvider::optimized_get_dock_usd_price` each time ?
This would mean I can get rid of all of the pallet's Storage, events entirely

Reply:
Yes.